rfcs/text/0009_screenshot_mode_service.md

```diff
@@ -0,0 +1,262 @@
+  rendering in Javascript. See "further examples" in this RFC about how this
+  alternative solution can still be leveraged.
+
+# Adoption strategy
+
+Integrating applications with this service is only going to be necessary when
+there are customizations needed to make the application more "reportable."
+That will be done by examining the default results of how applications perform
+for screenshot capture mode, and tuning the app code for performance as needed.
+
+Some of the first steps would be to have the Reporting services team hoist some
+code out of the screenshot pipeline, and start adding the `screenshots`
+variable to the URL as a query string parameter. Then, parts of the Reporting
+code can be taken apart a bit. The areas that need customization for screenshot
+capture will no longer rely on the Reporting plugin to inject those
+customizations. The applications will check the Screenshot Mode Service and
+inject the specialized customizations as needed.
+
+Having the Screenshot mode service would be a means towards fixing a few bugs
+with Reporting rendering. The bug fixes will involve adopting the Screenshot
+Mode Service throughout applications: especially in the Dashboard application.
+Those fixes will be driven by the Reporting Services team initially.
+Eventually, the Dashboard application could feature a few examples in the code
+on how to integrate with the Screenshot Mode Service.
+
+Further adoption of this service would happen as we look for ways to improve
+the performance and quality of Reporting services throughout Kibana. Those
+fine-tuning adoptions can be driven by the Reporting services team, or
+application teams, depending on bandwidth available.
+
+# How we teach this
+
+Not every Kibana developer will need to understand the screenshot mode service
+to be effective at day-to-day work, but every Kibana developer should be at
+least aware of its existence. That way, they can understand it when the time
+comes to specialize their application for Reporting. As long as developers are
+aware of its existence, they could understand it by reading the code and trying
+the test Kibana plugin (if it exists).
+
+# Further examples
+
+- Applications can also use screenshot context to customize the way they load.
+  An example is Toast Notifications: by default they auto-dismiss themselves
+  after 30 seconds or so. That makes sense when there is a human there to
+  notice the message, read it and remember it. But if the page is loaded for
+  capturing a screenshot, the toast notifications should never disappear. The
+  message in the toast needs to be part of the screenshot for its message to
+  mean anything, so it should not force the screenshot capture tool to race
+  against the toast timeout window.
+- Avoid collection and sending of telemetry from the browser when page is
+  loaded for screenshot capture.
+- Turn off autocomplete features and auto-refresh features that weigh on
```

Comment:
Can it be disabled by the `Screenshot Mode Service` explicitly via ui settings or plugin API?

Reply:
The plan for what the `Screenshot Mode Service` will have the applications and plugins turn off their own features as needed for more optimal screen capture. We're purposefully not having the service do anything automatically - I'm not sure that would even be possible as the service is a plugin that applications have to explicitly integrate with.